File: app/Models/Media.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Notifications\Notifiable;
use Laravel\Sanctum\HasApiTokens;
use Spatie\MediaLibrary\HasMedia;
use Spatie\MediaLibrary\InteractsWithMedia;



use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Support\Facades\Storage;

class Media extends Model implements HasMedia
{
    protected $table = 'media';
    use HasApiTokens, HasFactory, Notifiable, InteractsWithMedia;
    protected $fillable = [
        'seller_id',
        'store_id',
        'title',
        'height',
        'width',
        'name',
        'extension',
        'type',
        'sub_directory',
        'size',
        'order_column',
        'model_type',
        'model_id',
        'file_name',
        'disk',
        'conversions_disk',
        'collection_name',
        'mime_type',
        'custom_properties',
        'generated_conversions',
        'responsive_images',
        'manipulations',
        'uuid',
        'object_url',
    ];

    public function registerMediaCollections(): void
    {
        $media_storage_settings = fetchDetails(StorageType::class, ['is_default' => 1], '*');
        $mediaStorageType = !$media_storage_settings->isEmpty() ? $media_storage_settings[0]->name : 'public';
        if ($mediaStorageType == 's3') {
            $this->addMediaCollection('media')->useDisk('s3');
        } else {
            $this->addMediaCollection('media')->useDisk('public');
        }
    }
}

File: seller_app/lib/features/profile/addProduct/models/media.dart
class Media {
  int? id;
  String? name;
  String? image;
  String? size;
  String? extension;
  String? type;
  String? subDirectory;
  String? relativePath;
  Media(
      {this.id,
      this.extension,
      this.name,
      this.image,
      this.size,
      this.type,
      this.subDirectory,
      this.relativePath});
  Media.fromJson(Map<String, dynamic> json) {
    id = json["id"] ?? 0;
    name = json["name"] ?? "";
    size = json["size"] ?? "";
    type = json["type"] ?? "";
    extension = json["extension"] ?? "";
    image = json["image"] ?? "";
    subDirectory = json["sub_directory"] ?? "";
    if (json.containsKey("relative_path")) {
      relativePath = json["relative_path"];
    } else {
      relativePath = "${subDirectory!}/${name!}";
    }
  }
}
